frontend/ui/widget/skimtowidget.lua

```diff
@@ -42,31 +42,42 @@ function SkimToWidget:init()
         }
     end
 
-    self.buttons_layout = {}
+    -- nil for default center full mode; "top" and "bottom" for compact mode
+    local skim_dialog_position = G_reader_settings:readSetting("skim_dialog_position")
```

Comment:
When you later use this as a flag/booleante, it's a bit odd to read.
You could use `local compact_mode = skim_dialog_position ~= nil` and `if compact_mode then` in the many places below.